hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/JDBCSource.java

```diff
@@ -0,0 +1,238 @@
+      dataFrameReader = dataFrameReader.option(Config.URL_PROP, properties.getString(Config.URL));
+      dataFrameReader = dataFrameReader.option(Config.USER_PROP, properties.getString(Config.USER));
+      dataFrameReader = dataFrameReader.option(Config.DRIVER_PROP, properties.getString(Config.DRIVER_CLASS));
+      dataFrameReader = dataFrameReader
+          .option(Config.RDBMS_TABLE_PROP, properties.getString(Config.RDBMS_TABLE_NAME));
+
+      if (!properties.containsKey(Config.PASSWORD)) {
+        if (properties.containsKey(Config.PASSWORD_FILE)) {
+          if (!StringUtils.isNullOrEmpty(properties.getString(Config.PASSWORD_FILE))) {
+            LOG.info("Reading password for password file {}", properties.getString(Config.PASSWORD_FILE));
+            FileSystem fileSystem = FileSystem.get(new Configuration());
+            passwordFileStream = fileSystem.open(new Path(properties.getString(Config.PASSWORD_FILE)));
+            byte[] bytes = new byte[passwordFileStream.available()];
+            passwordFileStream.read(bytes);
+            dataFrameReader = dataFrameReader.option(Config.PASSWORD_PROP, new String(bytes));
+          } else {
+            throw new IllegalArgumentException(
+                String.format("%s property cannot be null or empty", Config.PASSWORD_FILE));
+          }
+        } else {
+          throw new IllegalArgumentException(String.format("JDBCSource needs either a %s or %s to connect to RDBMS "
+              + "datasource", Config.PASSWORD_FILE, Config.PASSWORD));
+        }
+      } else if (!StringUtils.isNullOrEmpty(properties.getString(Config.PASSWORD))) {
+        dataFrameReader = dataFrameReader.option(Config.PASSWORD_PROP, properties.getString(Config.PASSWORD));
+      } else {
+        throw new IllegalArgumentException(String.format("%s cannot be null or empty. ", Config.PASSWORD));
+      }
+//      if (properties.containsKey(Config.EXTRA_OPTIONS)) {
+//        if (!StringUtils.isNullOrEmpty(properties.getString(Config.EXTRA_OPTIONS))) {
+//          LOG.info("Setting {}", Config.EXTRA_OPTIONS);
+//          String[] options = properties.getString(Config.EXTRA_OPTIONS).split(",");
+//          for (String option : options) {
+//            if (!StringUtils.isNullOrEmpty(option)) {
+//              String[] kv = option.split("=");
+//              if (kv.length == 2) {
+//                dataFrameReader = dataFrameReader.option(kv[0], kv[1]);
+//                LOG.info("{} = {} has been set for JDBC pull ", kv[0], kv[1]);
+//              } else {
+//                LOG.warn("Option {} not set because it does not have a value", kv[0], new IllegalArgumentException(
+//                    String.format(" %s should have a corresponding value separated by \"=\"", kv[0])));
+//              }
+//            }
+//          }
+//        }
+//      }
+
+      properties.entrySet().stream().forEach(i-> System.out.println(properties.get(i)));
```

Comment:
System.out.println => LOG.info?

Reply:
That was for test purposes. The PR is not yet fully done. Work is still in progress